compiler/src/iree/compiler/Codegen/Common/CPU/CPUMaterializeEncodingPass.cpp

```diff
@@ -372,13 +370,28 @@ chooseMatmulTile(ArrayRef<TileMxNxK> enumeratedTiles, int64_t matmulNarrowM,
   SmallVector<RatedTileMxNxK> ratedTiles;
   ratedTiles.reserve(enumeratedTiles.size());
   int64_t bestPaddingPenalty = INT64_MAX;
+
+  auto roundUpToNextPowerOf2 = [](int64_t n) -> int64_t {
+    if (n <= 0)
+      return 1;
+    int64_t power = 1;
+    while (power < n)
+      power *= 2;
+    return power;
+  };
+
+  auto calculateUpperBound = [&](int64_t narrowValue) {
+    return narrowValue ? roundUpToNextPowerOf2(narrowValue)
+                       : hostDefinedUpperBound.getInt();
+  };
```

Comment:
I'd rather not have a lambda with an implicit capture for just avoiding writing twice this single ?: statement. Also, `calculateUpperBound` is not descriptive, but we dont need to argue about names if we dont have this lambda.

Reply:
this lambda is inlined and removed.